executor/api/rest/client.go

```diff
@@ -139,24 +142,26 @@ func (smc *JSONRestClient) getMetricsRoundTripper(modelName string, service stri
 
 func (smc *JSONRestClient) addHeaders(req *http.Request, m map[string][]string) {
 	for k, vv := range m {
-		for _, v := range vv {
-			req.Header.Set(k, v)
+		if _, ok := headersIgnore[k]; !ok {
+			for _, v := range vv {
+				req.Header.Set(k, v)
```

Comment:
If `vv` is a list of values, won't [`req.Header.Set`](https://golang.org/pkg/net/http/#Header.Set) override each previous entry and just leave the latest one? We may need to use [`req.Header.Add`](https://golang.org/pkg/net/http/#Header.Add) instead.

Reply:
Yes. Change to Add

Reply:
Does it work with the first one? You've got tests in there, so I guess it does work!